app/models/post.rb

```diff
@@ -1,3 +1,13 @@
 class Post < ApplicationRecord
-  belongs_to :user
+  belongs_to :author, class_name: 'User', counter_cache: :post_counter
+  has_many :like
+  has_many :comment
+
+  def top_5_comment
+    comment.order(created_at: :desc).limit(5)
+  end
+
+  def increment_post_counter
+    author.increment!(:post_counter)
+  end
```

Comment:
### **Optional**
- [ ] Please use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#increment_post_counter` method every time a new post has been saved to the database, then make the `increment_post_counter` method private.